server/channels/api4/user.go

```diff
@@ -1539,6 +1539,11 @@ func updateUserActive(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if user.AuthService == model.UserAuthServiceLdap {
+		c.Err = model.NewAppError("updateUserActive", "api.user.update_active.cannot_modify_status_when_user_is_managed_by_ldap.app_error", nil, "userId="+c.Params.UserId, http.StatusUnauthorized)
```

Comment:
Nit: this should be `http.StatusBadRequest`. Unauthorized means that the user doesn't have credentials to perform the action.
```suggestion
		c.Err = model.NewAppError("updateUserActive", "api.user.update_active.cannot_modify_status_when_user_is_managed_by_ldap.app_error", nil, "userId="+c.Params.UserId, http.StatusBadRequest)
```

Reply:
I was hesitant on which one to use, at the end I went for unauthorized, since IMO we are not "authorizing" the admin to do it via API. But 0/5 to keep it as Unauthorized. thoughts @harshilsharma63 @hanzei ?

Reply:
1/5 that 403 is the correct one as it makes it clear that the syntax of the request was valid, but the semantics are wrong. 400 is also fine.

Reply:
yes, we are in a gray area here for the correct status code. After thinking it again, I am 3/5 to keep it as unauthorized. The error message will give more clues of why is not allowed to modify the user and makes more sense than a malformed request. Unless @devinbinnie has a stronger oppinion on it, I'd like to keep it as 403.

Reply:
> The 401 (Unauthorized) status code indicates that the request has not
   been applied because it lacks valid authentication credentials for
   the target resource.  The server generating a 401 response MUST send
   a WWW-Authenticate header field ([Section 4.1](https://datatracker.ietf.org/doc/html/rfc7235#section-4.1)) containing at least one
   challenge applicable to the target resource.

> If the request included authentication credentials, then the 401
   response indicates that authorization has been refused for those
   credentials.  The user agent MAY repeat the request with a new or
   replaced Authorization header field ([Section 4.2](https://datatracker.ietf.org/doc/html/rfc7235#section-4.2)).  If the 401
   response contains the same challenge as the prior response, and the
   user agent has already attempted authentication at least once, then
   the user agent SHOULD present the enclosed representation to the
   user, since it usually contains relevant diagnostic information.

https://datatracker.ietf.org/doc/html/rfc7235#section-3.1

It doesn't seem like that case applies here. The credentials are fine, right?

Reply:
Yes, went and took a second look and @hanzei is right, 403 (`http.StatusForbidden`) is what we actually want. Thanks for clarifying :)

Reply:
changed